WcaOnRails/app/models/competition.rb

```diff
@@ -35,6 +35,33 @@ def cleanup_event_specs
     self.eventSpecs ||= ""
   end
 
+  before_validation :create_id_and_cell_name
```

Comment:
There's actually a test just like this in the competition controller spec. See https://github.com/jfly/worldcubeassociation.org/blob/a5eca69ea816ba5694c1f040aae76f5415743c03/WcaOnRails/spec/controllers/competitions_controller_spec.rb#L45-L50. I'll add a test to the competition_spec as well, though. Doesn't cost us anything.